code/src/components/MovieDetails.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { useEffect, useState } from 'react';
+import { useParams, Link } from 'react-router-dom';
+import { BackIcon } from './BackIcon';
+
+import { MOVIEDETAILS_URL } from '../utils/urls';
+
+const MovieDetails = () => {
+  const [movieDetails, setMovieDetails] = useState([]);
+  const [hasError, setHasError] = useState(false);
+
+  const { id } = useParams();
+
+  useEffect(() => {
+    fetch(MOVIEDETAILS_URL(id))
+      .then((res) => res.json())
+      .then((data) => {
+        if (data.id) {
+          setMovieDetails(data);
+        } else {
+          setHasError(true);
+        }
+      })
+      .catch(() => setHasError(true));
+  }, [id]);
+
+  if (hasError) {
+    return (
+      <div>
+        <h2 className='error-message'>Movie not found</h2>
+        <Link to='/' className='backLink'>
+          <BackIcon /> Movies
+        </Link>
+      </div>
+    );
+  }
+
+  return (
+    <div className='detail-page'>
+      <Link to='/' className='backLink'>
+        <BackIcon /> Movies
+      </Link>
+      <div
+        className='backdrop'
+        style={{
+          backgroundImage: movieDetails.backdrop_path
```

Comment:
This is covering the situation where backdrop_path doesn't exist for some reason, right? It's really nice that you've considered this. It makes your code very robust I think.